Experiment_Launcher_code/ExperimentLauncher.py

```diff
@@ -6,27 +15,34 @@ def main():
     experiment_gui = ExperimentGUI()
     experiment_gui.setup_gui()
 
-    # After the GUI is closed, get the settings using the get_settings method
-    settings = experiment_gui.on_start_clicked()
-
-    if settings:
-        # Extract the necessary parameters from the settings
-        experiment_name = settings.get('experiment_name')
-        comport_name = settings.get('comport_name')
-        num_trials = settings.get('num_trials')
-        return_time = settings.get('return_time')
-        decision_time = settings.get('decision_time')
-        opponent_type = settings.get('opponent_type')
-        opponent1_strategy = settings.get('opponent1_strategy')
-        opponent2_strategy = settings.get('opponent2_strategy')
-        opponent1_probability = settings.get('opponent1_probability')
-        opponent2_probability = settings.get('opponent2_probability')
+    # After the GUI is closed, get the settings using the appropriate methods
+    if experiment_gui.is_data_valid():
+        comport_name = experiment_gui.get_com_port()
+        experiment_parameters = experiment_gui.get_experiment_parameters()
+        opponent_configuration = experiment_gui.get_opponent_configuration()
+
+        # Instantiate software components
+        video_analyzer = Video_Analyzer("1780")
+        reward_manager = RewardManager(comport_name)
+
+        if opponent_configuration.get("opponent1_type") == OpponentType.MOUSE:
```

Comment:
i assume youve moved this part from the experimentlauncher to the manager as you plan to incorporate further changes  regarding the types of learners and strategies